code/upstream-integration-tests/src/androidTest/java/com/adobe/marketing/mobile/integration/ConfigOverridesIntegrationTests.kt

```diff
@@ -0,0 +1,357 @@
+    private val VALID_DATASET_ID_CONFIGURED_AS_OVERRIDE = "6515e1dbfeb3b128d19bb1e4"
+    private val VALID_DATASET_ID_NOT_CONFIGURED_AS_OVERRIDE = "6515e1f6296d1e28d3209b9f"
+    private val VALID_RSID_CONFIGURED_AS_OVERRIDE = "mobile5.e2e.rsid2"
+    private val VALID_RSID_NOT_CONFIGURED_AS_OVERRIDE = "mobile5e2e.rsid3"
+
+    @JvmField
+    @Rule
+    var rule: RuleChain = RuleChain
+        .outerRule(TestHelper.LogOnErrorRule())
+        .around(TestHelper.SetupCoreRule())
+        .around(TestHelper.RegisterMonitorExtensionRule())
+
+    @Before
+    @Throws(Exception::class)
+    fun setup() {
+        println("Environment var - Edge Network environment: $edgeEnvironment")
+        println("Environment var - Edge Network location hint: $edgeLocationHint")
+        ServiceProvider.getInstance().networkService = realNetworkService
+
+        // Set environment file ID for specific Edge Network environment
+        MobileCore.configureWithAppID(getEnvironmentFileID(edgeEnvironment))
+        val latch = CountDownLatch(1)
+        MobileCore.registerExtensions(listOf(Edge.EXTENSION, Identity.EXTENSION)) {
+            latch.countDown()
+        }
+        latch.await()
+
+        // Set Edge location hint if one is set for the test suite
+        if (edgeLocationHint.isNotEmpty()) {
+            print("Setting Edge location hint to: $edgeLocationHint")
+            Edge.setLocationHint(edgeLocationHint)
+        }
+        else {
+            print("No preset Edge location hint is being used for this test.")
+        }
+
+        resetTestExpectations()
+    }
+
+    @After
+    fun tearDown() {
+        realNetworkService.reset()
+        TestHelper.resetCoreHelper()
+    }
+
+    @Test
+    fun testSendEvent_withValidDatastreamIdOverride_receivesExpectedNetworkResponse() {
+        val interactNetworkRequest = TestableNetworkRequest(createInteractURL(locationHint = edgeLocationHint), HttpMethod.POST)
+
+        realNetworkService.setExpectationForNetworkRequest(interactNetworkRequest, expectedCount = 1)
+
+        val experienceEvent = ExperienceEvent.Builder()
+            .setXdmSchema(mapOf("xdmtest" to "data"))
+            .setData(mapOf("data" to mapOf("test" to "data")))
+            .setDatastreamIdOverride(VALID_DATASTREAM_ID_OVERRIDE)
+            .build()
+
+        Edge.sendEvent(experienceEvent) {}
+
+        realNetworkService.assertAllNetworkRequestExpectations()
+        val matchingResponses = realNetworkService.getResponsesFor(interactNetworkRequest)
+
+        assertEquals(1, matchingResponses?.size)
+        assertEquals(200, matchingResponses?.firstOrNull()?.responseCode)
+    }
+
+    @Test
+    fun testSendEvent_withInvalidDatastreamIdOverride_receivesErrorResponse() {
+        val interactNetworkRequest = TestableNetworkRequest(createInteractURL(locationHint = edgeLocationHint), HttpMethod.POST)
+
+        realNetworkService.setExpectationForNetworkRequest(interactNetworkRequest, expectedCount = 1)
```

Comment:
For all the tests that assert on the handle we need to set the expectation for that event handle type:
```
expectEdgeEventHandle(TestConstants.EventSource.ERROR_RESPONSE_CONTENT, 1)
```